Libraries/LibC/unistd.cpp

```diff
@@ -568,6 +568,12 @@ int umount(const char* mountpoint)
     __RETURN_WITH_ERRNO(rc, rc, -1);
 }
 
+int realpath(const char* pathname, char* buffer)
+{
+    int rc = syscall(SC_realpath, pathname, buffer, sizeof(buffer));
```

Comment:
No, `sizeof(buffer)` is just `4`.

Reply:
Right. And if `buffer` is null, then `realpath` should `malloc` a buffer.

Reply:
Reading the POSIX spec again, `buffer` should either point to a `PATH_MAX`-sized buffer or be null, in which case this function should malloc a `PATH_MAX`-sized buffer; in any case the kernel gets a `PATH_MAX`-sized buffer, so there's no point in passing the size to the kernel here.

(Or `PATH_MAX` could be undefined, as it is on the Hurd...)

Reply:
Done.